src/wallet/wallet.cpp

```diff
@@ -124,13 +124,21 @@ bool CWallet::AddKeyPubKey(const CKey& secret, const CPubKey &pubkey)
     if (HaveWatchOnly(script))
         RemoveWatchOnly(script);
 
-    if (!fFileBacked)
+    if (!fFileBacked) {
+        RefreshUnspents();
         return true;
+    }
+
     if (!IsCrypted()) {
-        return CWalletDB(strWalletFile).WriteKey(pubkey,
+        if(!CWalletDB(strWalletFile).WriteKey(pubkey,
                                                  secret.GetPrivKey(),
-                                                 mapKeyMetadata[pubkey.GetID()]);
+                                                 mapKeyMetadata[pubkey.GetID()]))
+        return false;
     }
+
+    // Refresh current UTXOs to check if some of them were already being tracked
+    // but marked as unspendable. Their info will be updated.
+    RefreshUnspents();
```

Comment:
When importing a private key of a previously imported public address, we need to go through the list of UTXOs to check if they were being tracked and marked as unspendable, and update its info.